### DISCUSSION TITLE
Configurable Polling Rates for Player Providers (Sonos)

### DISCUSSION THREAD

**Discussion by alexzeitgeist** [music-assistant/support#2129] [Feature requests and ideas]
**Current Behavior:** The Music Assistant server continuously polls Sonos players every two minutes to verify their availability.

**Concern:** Constant polling might:

- Prevent players from entering low-power mode.
- Result in unnecessary energy consumption.

**Contexts:**

- The players are constantly connected and powered on.
- No devices are currently using the Music Assistant server (for example no client connections via the standalone web interface).

**Observation:** In both contexts mentioned above, continuous polling appears unnecessary.

**Suggestion**: Allow users to adjust the polling frequency for each player provider, ideally within the 'Advanced' settings. Additionally, consider implementing a feature to temporarily cease polling when no devices are connected to the Music Assistant server.

Thanks,
Alex

**Comment by marcelveldt** [comment]
Polling is needed due to the unreliability of the UPNP callbacks. Otherwise there is no good way to learn if the player is actually alive.
But, for all players running the S2 software we can use the new API which is much much better than the crappy UPNP shit so I intend to release an option soon to switch to the S2 mode. It requires a bit of refactoring but I think that is worth the effort.

Also, I do not think that a sonos speaker is prevented from going into low power mode.
It should go into low power mode if its not playing for X seconds, its webserver should stay alive.
